src/lib/db.ts

```diff
@@ -0,0 +1,83 @@
+import * as firebase from "firebase";
+import { AppMode } from "../models/stores";
+
+export type IDBConnectOptions = IDBAuthConnectOptions | IDBNonAuthConnectOptions;
+export interface IDBAuthConnectOptions {
+  appMode: "authed";
+  rawFirebaseToken: string;
+}
+export interface IDBNonAuthConnectOptions {
+  appMode: "dev" | "test";
+}
+
+export class DB {
+  private appMode: AppMode;
+  private firebaseUser: firebase.User | null = null;
+
+  public get isConnected() {
+    return this.firebaseUser !== null;
+  }
+
+  public connect(options: IDBConnectOptions) {
+    return new Promise<boolean>((resolve, reject) => {
+      if (this.isConnected) {
+        reject("Already connected to database!");
+      }
+
+      // check for already being initialized for tests
+      if (firebase.apps.length === 0) {
+        firebase.initializeApp({
+          apiKey: "AIzaSyBKwTfDSxKRSTnOaAzI-mUBN78LiI2gM78",
```

Comment:
Maybe it doesn't matter, but these seem like they should be named constants or configuration parameters.